library/oneview_logical_switch_group.py

```diff
@@ -0,0 +1,206 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+from ansible.module_utils.basic import *
+from hpOneView.oneview_client import OneViewClient
+from hpOneView.common import resource_compare
+
+DOCUMENTATION = '''
+---
+module: oneview_logical_switch_group
+short_description: Manage OneView Logical Switch Group resources.
+description:
+    - "Provides an interface to manage Logical Switch Group resources. Can add, update, remove."
+requirements:
+    - "python >= 2.7.9"
+    - "hpOneView"
+author: "Gustavo Hennig (@GustavoHennig)"
+options:
+    config:
+        description:
+            - Path to a .json configuration file containing the OneView client configuration.
+        required: true
+    state:
+        description:
+            - Indicates the desired state for the Logical Switch Group resource.
+              'present' will ensure data properties are compliant to OneView.
+              'absent' will remove the resource from OneView, if it exists.
+        choices: ['present', 'absent']
+        required: true
+    data:
+        description:
+            - List with Logical Switch Group properties and its associated states
+        required: true
+notes:
+    - "A sample configuration file for the config parameter can be found at:
+       https://github.com/HewlettPackard/oneview-ansible/blob/master/examples/oneview_config-rename.json"
+'''
+
+EXAMPLES = '''
+- name: Create a Logical Switch Group
+  oneview_logical_switch_group:
+    config: "{{ config }}"
+    state: present
+    data:
+        type: "logical-switch-group"
+        name: "OneView Test Logical Switch Group"
+        switchMapTemplate:
+            switchMapEntryTemplates:
+                - logicalLocation:
+                    locationEntries:
+                       - relativeValue: 1
+                         type: "StackingMemberId"
+                  # You can choose either permittedSwitchTypeName or permittedSwitchTypeUri to inform the Switch Type
+                  permittedSwitchTypeName: 'Cisco Nexus 50xx'
+                  permittedSwitchTypeUri: '/rest/switch-types/2f36bc8f-65d8-4ea2-9300-750180402a5e'
+  delegate_to: localhost
+
+- name: Update the Logical Switch Group
+  oneview_logical_switch_group:
+    config: "{{ config }}"
+    state: present
+    data:
+        type: "logical-switch-group"
+        name: "OneView Test Logical Switch Group"
+        newName: "Test Logical Switch Group"
+        switchMapTemplate:
+            switchMapEntryTemplates:
+                - logicalLocation:
+                    locationEntries:
+                       - relativeValue: 1
+                         type: "StackingMemberId"
+                  permittedSwitchTypeUri: '{{ switch_type_uri }}'
```

Comment:
I would prefer display here the URI '/rest/switch-types/2f36bc8f-65d8-4ea2-9300-750180402a5e' rather than the variable name.